client/src/components/Block.tsx

```diff
@@ -1,49 +1,101 @@
 import React, { useContext } from "react";
 import "../styling/Maze.css";
 import { PageContext } from "../PageProvider";
+import { SetState, Maze, Point } from "../utilities/types";
+import { updateMaze } from "../utilities/utilities";
 
 type BlockProps = {
   blockType: number;
   mazeSize: number;
+  maze: Maze;
+  rowIndex: number;
+  colIndex: number;
+  setMaze: SetState<Maze>;
 };
 
-const Block: React.FC<BlockProps> = ({ blockType, mazeSize }) => {
+const Block: React.FC<BlockProps> = ({ blockType, mazeSize, maze, rowIndex, colIndex, setMaze }) => {
```

Comment:
anything to do with maze, should be put into a MazeContext and subscribed here via a provider

Reply:
added Maze Provider for Maze context